htlcswitch/test_utils.go

```diff
@@ -0,0 +1,518 @@
+		OurBalance:             aliceAmount,
+		TheirBalance:           bobAmount,
+		OurCommitTx:            aliceCommitTx,
+		OurCommitSig:           bytes.Repeat([]byte{1}, 71),
+		FundingOutpoint:        prevOut,
+		OurMultiSigKey:         aliceKeyPub,
+		TheirMultiSigKey:       bobKeyPub,
+		FundingWitnessScript:   witnessScript,
+		LocalCsvDelay:          csvTimeoutAlice,
+		RemoteCsvDelay:         csvTimeoutBob,
+		TheirCurrentRevocation: bobRevokeKey,
+		LocalElkrem:            ourElkrem,
+		RemoteElkrem:           &elkrem.ElkremReceiver{},
+		TheirDustLimit:         bobDustLimit,
+		OurDustLimit:           aliceDustLimit,
+		Db:                     dbAlice,
+	}
+	bobChannelState := &channeldb.OpenChannel{
+		IdentityPub:            bobKeyPub,
+		ChanID:                 prevOut,
+		ChanType:               channeldb.SingleFunder,
+		IsInitiator:            false,
+		StateHintObsfucator:    obsfucator,
+		OurCommitKey:           bobKeyPub,
+		TheirCommitKey:         aliceKeyPub,
+		Capacity:               channelCapacity,
+		OurBalance:             bobAmount,
+		TheirBalance:           aliceAmount,
+		OurCommitTx:            bobCommitTx,
+		OurCommitSig:           bytes.Repeat([]byte{1}, 71),
+		FundingOutpoint:        prevOut,
+		OurMultiSigKey:         bobKeyPub,
+		TheirMultiSigKey:       aliceKeyPub,
+		FundingWitnessScript:   witnessScript,
+		LocalCsvDelay:          csvTimeoutBob,
+		RemoteCsvDelay:         csvTimeoutAlice,
+		TheirCurrentRevocation: aliceRevokeKey,
+		LocalElkrem:            theirElkrem,
+		RemoteElkrem:           &elkrem.ElkremReceiver{},
+		TheirDustLimit:         aliceDustLimit,
+		OurDustLimit:           bobDustLimit,
+		Db:                     dbBob,
+	}
+
+	cleanUpFunc := func() {
+		os.RemoveAll(bobPath)
+		os.RemoveAll(alicePath)
+	}
+
+	aliceSigner := &MockSigner{aliceKeyPriv}
+	bobSigner := &MockSigner{bobKeyPriv}
+
+	notifier := &MockNotifier{}
+
+	channelAlice, err := lnwallet.NewLightningChannel(aliceSigner, nil,
+		notifier, firstChannelState)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	channelBob, err := lnwallet.NewLightningChannel(bobSigner, nil,
+		notifier, bobChannelState)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return channelAlice, channelBob, cleanUpFunc, nil
+}
+
+// Cluster is used for managing the created cluster of 3 hops.
+type Cluster struct {
+	aliceServer      *MockServer
+	aliceHtlcManager *htlcManager
+
+	firstBobHtlcManager  *htlcManager
+	bobServer            *MockServer
+	secondBobHtlcManager *htlcManager
+
+	carolHtlcManager *htlcManager
+	carolServer      *MockServer
+
+	aliceRegistry *MockInvoiceRegistry
+
+	firstChannelCleanup  func()
+	secondChannelCleanup func()
+}
+
+// MakeCarolToAlicePayment makes payment from Carol to Alice over Bob which
+// includes:
+// * creating the invoice and adding it on Alice side.
+// * create Carol payment request.
+// * handle user request by Carol HTLC switch.
+func (c *Cluster) MakeCarolToAlicePayment(amount btcutil.Amount,
```

Comment:
Additionally, it attempts to pack to a heap of logic that is specific to the way the tests currently operate. Instead, it should be broken down into a series of modular functions which can them be composed within tests in order to achieve the same results. 

A similar comment applies to any of the `MakeXToXPayment` methods.